var/spack/repos/builtin/packages/neovim/package.py

```diff
@@ -25,26 +26,27 @@ class Neovim(CMakePackage):
     version('0.2.0', sha256='72e263f9d23fe60403d53a52d4c95026b0be428c1b9c02b80ab55166ea3f62b5')
 
     depends_on('cmake@3.0:', type='build')
+    depends_on('git@1.8.4:', type='build', when='@stable,master')
     depends_on('pkgconfig', type='build')
     depends_on('gettext', type=('build', 'link'))
-    depends_on('lua@5.1:5.2', type=('build', 'link'))
+    depends_on('lua@5.1.0:5.1.9', type=('build', 'link'))
     depends_on('lua-lpeg', type='link')
     depends_on('lua-mpack', type='link')
     depends_on('lua-bitlib', type='link')
     depends_on('libuv', type='link')
-    depends_on('libuv@1.28:', type='link', when='@0.4:')
+    depends_on('libuv@1.28:', type='link', when='@0.4:,master,stable')
     depends_on('jemalloc', type='link')
     depends_on('libtermkey', type='link')
-    depends_on('libtermkey@0.18:', type='link', when='@0.3.4:')
-    depends_on('libvterm@0.0', type='link', when='@:0.3')
-    depends_on('libvterm@0.1:', type='link', when='@0.4:')
+    depends_on('libtermkey@0.18:', type='link', when='@0.3.4:,master,stable')
+    depends_on('libvterm@0.0.0', type='link', when='@0.2.0:0.3')
+    depends_on('libvterm@0.1:', type='link', when='@0.4:,master,stable')
     depends_on('unibilium', type='link')
-    depends_on('unibilium@:1.2.0', type='link', when='@:0.2.0')
-    depends_on('unibilium@2.0:', type='link', when='@0.4:')
+    depends_on('unibilium@:1.2.0', type='link', when='@0.2.0')
```

Comment:
why the additional restriction?

Reply:
which restriction exactly?

Reply:
which restriction exactly?

Reply:
You switched from "everything up to 0.2.0" to "exactly 0.2.0"

Reply:
Yes - I tried all lower version than 0.2.0 and they all don't compile - and none of the are mentioned by hash in the file. Happy to include for everything lower.

Reply:
ah, if you know that the older versions do not build then it's good to remove it. In general we try to keep the dependencies as general as possible. I.e. while the package does not contain information about older versions right now, adding new ones can happen anytime, and we try to minimise the overhead and let the concretiser do as much work as possible. 

In this case I'd mention the additional restriction in the commit message, including that you tried older versions and they didn't work. (You could also add the older versions)

Reply:
ok - will add it in the commit message. And I agree that while generally I am all in favor of fixing also for older versions ... in this case I will abstain as this is not likely to be a dependency of something else as it is an editor ...